fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -862,7 +865,9 @@ public void analyze(TQueryOptions tQueryOptions) throws UserException, Interrupt
                     context.getForwardedStmtId());
         }
 
-        parseByLegacy();
+        if (parsedStmt == null) {
+            parseByLegacy();
+        }
```

Comment:
why add `parsedStmt == null`? we already check `parsedStmt == null` in `parseByLegacy()`